### PR TITLE
Sample license addition for review, before I add them in every file.

### PR DESCRIPTION
**Pull request by vext01**
Here's (to begin with) a sample of what the license looks like. I wanted to check it's right before adding it in lots of places.

Wrapped using `gq` in vim.

OK to continue?

Cheers

### CODE REVIEW

**Comment by ltratt** [comment]
And I.

**Comment by vext01** [comment]
I think this is the lot.

**Comment by vext01** [comment]
I didn't add licenses to Makefiles. Is that right? Seems excessive...

File: README.md [hunk 1]
```diff
@@ -433,6 +433,9 @@ Please make sure you understand the implications of this.
 
 ## Licenses
 
+The main bulk of Krun is licensed under the UPL license, which is found at the
```
Comment (ltratt) [inline README.md]: This section is verbose and confusing. I'd suggest either removing it or saying "Krun is licensed under the UPL. However, the nbody benchmark is published under a ..." or similar.

**Comment by ltratt** [comment]
Agreed, makefiles is one step too far (except in weird cases... which we don't have). I just have one comment.

File: README.md (hunk 1 above)
Comment (vext01) [inline README.md]: Made it shorter.

**Comment by vext01** [comment]
The benchmark is provided as an example.

**Comment by vext01** [comment]
Wait, let me re-word slightly.

**Comment by ltratt** [comment]
I don't think it's an example: it's a benchmark we use. I think we can just say "the nbody benchmark is...".

**Comment by vext01** [comment]
No, it really is only an example in this repo.

**Comment by ltratt** [comment]
Ah, OK. But I agree: "example" isn't important to the license.

OK, I think I'm happy. Time to squash?

**Comment by vext01** [comment]
I think so.

**Comment by ltratt** [comment]
OK, squash away.

**Comment by vext01** [comment]
Big splat!

**Comment by ltratt** [comment]
OK, thanks, will wait for the tests then merge.

**Comment by vext01** [comment]
I suppose we should do the same on the warmup experiment itself (tomorrow)

**Comment by ltratt** [comment]
Good point.

**Comment by vext01** [comment]
There we already have in the README:
```
All original files are under the following MIT license
```

Is that still OK?

**Comment by vext01** [comment]
(Obviously, add the header to all source files)

**Comment by ltratt** [comment]
We should change MIT -> UPL IMHO, but otherwise yes.

**Pull request merged**